arelle/Locale.py

```diff
@@ -516,10 +549,12 @@ def format_picture(conv, value, picture):
 
     if isinstance(value, float):
         value = Decimal.from_float(value)
-    elif isinstance(value, STR_NUM_TYPES):
+    elif isinstance(value, STR_NUM_TYPES) and not isinstance(value, Fraction):
```

Comment:
The problem here is that Decimal does not expect a Fraction, so I have to exclude the fraction anyway and isolate its test.

Reply:
This is an annoying limitation of the type checker. It would be better to update the condition to explicitly check for what we want:
```python
    elif isinstance(value, (str, int)):
        value = Decimal(value)
    elif isinstance(value, Fraction):
        value = Decimal(float(value))
```